File: Demo/Objective_C_Demo/ViewController/BottomBlankSpaceViewController.m
//
//  BottomBlankSpaceViewController.m
//  IQKeyboard
//
//  Created by Iftekhar on 15/10/14.
//  Copyright (c) 2014 Iftekhar. All rights reserved.
//

#import "BottomBlankSpaceViewController.h"
#import "IQKeyboardManager.h"

@interface BottomBlankSpaceViewController () <UIPopoverPresentationControllerDelegate>

@end

@implementation BottomBlankSpaceViewController
{
    IBOutlet UISwitch *switchPreventShowingBottomBlankSpace;
}

-(void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];

    switchPreventShowingBottomBlankSpace.on = [[IQKeyboardManager sharedManager] preventShowingBottomBlankSpace];
}

- (IBAction)preventSwitchAction:(UISwitch *)sender
{
    [[IQKeyboardManager sharedManager] setPreventShowingBottomBlankSpace:switchPreventShowingBottomBlankSpace.on];
}

-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.identifier isEqualToString:@"SettingsNavigationController"])
    {
        segue.destinationViewController.modalPresentationStyle = UIModalPresentationPopover;
        segue.destinationViewController.popoverPresentationController.barButtonItem = sender;
        
        CGFloat heightWidth = MAX(CGRectGetWidth([[UIScreen mainScreen] bounds]), CGRectGetHeight([[UIScreen mainScreen] bounds]));
        segue.destinationViewController.preferredContentSize = CGSizeMake(heightWidth, heightWidth);
        segue.destinationViewController.popoverPresentationController.delegate = self;
    }
}

- (UIModalPresentationStyle)adaptivePresentationStyleForPresentationController:(UIPresentationController *)controller
{
    return UIModalPresentationNone;
}

-(void)prepareForPopoverPresentation:(UIPopoverPresentationController *)popoverPresentationController
{
    [self.view endEditing:YES];
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    return YES;
}

- (BOOL)shouldAutorotate
{
    return YES;
}


@end

File: Demo/Swift_Demo/ViewController/BottomBlankSpaceViewController.swift
//
//  BottomBlankSpaceViewController.swift
//  IQKeyboardManager
//
//  Created by InfoEnum02 on 20/04/15.
//  Copyright (c) 2015 Iftekhar. All rights reserved.
//

import UIKit
import IQKeyboardManagerSwift

class BottomBlankSpaceViewController : UIViewController, UIPopoverPresentationControllerDelegate {
    
    @IBOutlet fileprivate var switchPreventShowingBottomBlankSpace : UISwitch!

    override func viewWillAppear(_ animated : Bool) {
        super.viewWillAppear(animated)
        
        switchPreventShowingBottomBlankSpace.isOn = IQKeyboardManager.sharedManager().preventShowingBottomBlankSpace
    }
    
    @IBAction func preventSwitchAction (_ sender: UISwitch!) {
        IQKeyboardManager.sharedManager().preventShowingBottomBlankSpace = sender.isOn
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        
        if let identifier = segue.identifier {
            
            if identifier == "SettingsNavigationController" {
                
                let controller = segue.destination
                
                controller.modalPresentationStyle = .popover
                controller.popoverPresentationController?.barButtonItem = sender as? UIBarButtonItem
                
                let heightWidth = max(UIScreen.main.bounds.width, UIScreen.main.bounds.height);
                controller.preferredContentSize = CGSize(width: heightWidth, height: heightWidth)
                controller.popoverPresentationController?.delegate = self
            }
        }
    }

    func adaptivePresentationStyle(for controller: UIPresentationController) -> UIModalPresentationStyle {
        return .none
    }

    func prepareForPopoverPresentation(_ popoverPresentationController: UIPopoverPresentationController) {
        self.view.endEditing(true)
    }

    override var shouldAutorotate : Bool {
        return true
    }
}
